categorize_notify/program.py

```diff
@@ -10,60 +10,86 @@
 
 import base64
 import time
+from datetime import datetime, timezone
 
 import autokitteh
 from autokitteh import google, openai, slack
 
 
 SLACK_CHANNELS = ["demos", "engineering", "ui"]
 
+processed_message_ids = set()
+start_time = datetime.now(timezone.utc)
+
 
 def on_http_get(event):
-    total_messages = None
     while True:
-        total_messages = _poll_inbox(total_messages)
+        _poll_inbox()
         time.sleep(10)
 
 
 @autokitteh.activity
-def _poll_inbox(prev_total: int):
+def _poll_inbox():
     gmail = google.gmail_client("my_gmail").users()
-    curr_total = gmail.getProfile(userId="me").execute()["messagesTotal"]
-    # Note: This is not meant to be a robust solution for handling email operations.
-    if prev_total and curr_total > prev_total:
-        new_email_count = curr_total - prev_total
-        message_ids = _get_latest_message_ids(gmail, new_email_count)
-        for message_id in message_ids:
-            _process_email(gmail, message_id)
+    token = None
+    current_message_ids = set()
+
+    while True:
+        query = "in:inbox -in:drafts"
+        if token:
+            gmail_list = gmail.messages().list(userId="me", q=query, pageToken=token)
+            results = gmail_list.execute()
+        else:
+            results = gmail.messages().list(userId="me", q=query).execute()
+
+        current_message_ids.update({msg["id"] for msg in results.get("messages", [])})
 
-    return curr_total
+        token = results.get("nextPageToken")
+        if not token:
+            break
+
+    new_message_ids = current_message_ids - processed_message_ids
+
+    for message_id in new_message_ids:
+        _process_email(gmail, message_id)
+
+    processed_message_ids.update(new_message_ids)
 
 
 def _process_email(gmail, message_id: str):
     message = gmail.messages().get(userId="me", id=message_id).execute()
+    email_timestamp = int(message["internalDate"]) / 1000
+    email_datetime = datetime.fromtimestamp(email_timestamp, timezone.utc)
+
+    if email_datetime < start_time:
```

Comment:
no chance of missing stuff here with same start_time but that were not read yet?